scripts/test/bs-wrapper.js

```diff
@@ -43,6 +45,49 @@ async function hasRunningBuild() {
 }
 
 function runTests() {
-  const [command, ...args] = process.argv.slice(2)
-  return spawnCommand(command, args)
+  return new Promise((resolve) => {
+    const [command, ...args] = process.argv.slice(2)
+
+    const child = spawn(command, args, {
+      stdio: ['inherit', 'pipe', 'pipe'],
+      env: { ...process.env, FORCE_COLOR: true },
+    })
+
+    let output = ''
+    let isKilled = false
+    let timeoutId
+
+    child.stdout.pipe(process.stdout)
+    child.stdout.on('data', onOutput)
+
+    child.stderr.pipe(process.stderr)
+    child.stderr.on('data', onOutput)
+
+    child.on('exit', (code) => {
+      resolve(!isKilled && code === 0)
+    })
+
+    function onOutput(data) {
+      output += data
+
+      if (hasUnrecoverableFailure(output)) {
+        killIt('unrecoverable failure')
+      }
+
+      clearTimeout(timeoutId)
+      setTimeout(() => killIt('no output timeout'), NO_OUTPUT_TIMEOUT)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏ 
```suggestion
      timeoutId = setTimeout(() => killIt('no output timeout'), NO_OUTPUT_TIMEOUT)
```

Reply:
Aah good catch, that's why it didn't work :D